### ISSUE TITLE
`TypeAlias` forward reference results in `Unknown`

### ISSUE THREAD

**Issue by Skynet0** [microsoft/pylance-release#3654]
<h3> Environment data </h3>

-   Language Server version: 2022.11.30
-   OS and version: linux x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: false
-   python.analysis.typeCheckingMode: strict

<h3> Code Snippet </h3>

```python
from typing import TypeAlias

Foo: TypeAlias = "Bar"

class Bar:
    pass

x: Foo
x = Bar()

```

<h3> Repro Steps </h3>

Copy-paste the above code and see Pylance's highlighting / hover type hints.

<h3> Expected behavior </h3>

The type hints are correct, and Pylance does not error. The code typechecks according to mypy 0.991, and forward references in explicit type aliases are supported in [PEP 613](https://peps.python.org/pep-0613/#forward-references)

<h3> Actual behavior </h3>

The type hints give red underlines. `Bar` is "Unknown".

![image](https://user-images.githubusercontent.com/12869082/202842882-05094550-d45b-4544-8425-e67d81be6fe3.png)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in an upcoming release.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2022.11.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221141-30-November-2022-PreRelease)
